src/lib/with-timing.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * A utility method to wrap a function, and log execution time.
+ */
+export async function withTiming<T>(
+	label: string,
+	fn: () => Promise<T> | T
+): Promise<T> {
```

Comment:
💭 I like the grouping, but it also introduces extra noise for "leaf" timings. E.g. when I'm wrapping a fetch call in `withTiming` I don't necessarily want to see the two log lines that coming with using `console.group()`. What do you think about adding an option to disable the grouping behavior?

Reply:
yea, I was grappling with this a little bit...

And are you referring to this doubling-up?

![CleanShot 2022-11-29 at 17 16 05@2x](https://user-images.githubusercontent.com/26389321/204660388-d36e0c98-8c4d-4de6-89a8-2ba36144a7de.png)

Reply:
Yeah, exactly 👍

Reply:
@BRKalow -> https://github.com/hashicorp/dev-portal/pull/1395/commits/bc5ed7ccf8fbea329c56976a0d0a5bff97aaf91c

```console
[middleware] /consul/docs: 0.242ms
[[productSlug]/docs/index::getStaticProps] (consul)
  [docs-view/server::loadStaticProps]
    [getTutorialMap] (http://localhost:3000/api/tutorials-map)
      wait  - compiling /api/tutorials-map (client and server)...
      event - compiled successfully in 53 ms (1077 modules)
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100): 118.432ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=2b5a0644-8b93-4dba-ac5c-7843de4b1289): 39.375ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=0af4d72f-51dd-4dc5-85e4-67c68030ce8d): 46.31ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=8bd94156-cf0a-4f88-b5f5-d2ffa57f6402): 39.629ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=e48563d4-00d1-4c0b-a5da-13e276ab119c): 38.701ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=75bb9296-0364-4ec4-8fbd-3930c9e947d6): 49.063ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=216b3270-fc89-4ad5-9053-603be0df67e8): 43.516ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=95a7e36f-44ac-4e39-9fb3-c25db26f4af9): 58.987ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=6d51c189-d38e-493f-b373-7e2dfe6bdbc5): 50.253ms
    [getTutorialMap] (http://localhost:3000/api/tutorials-map): 1.142s
  [docs-view/server::loadStaticProps]: 1.234s
  [docs-view/server::prepareNavDataForClient]
    [getTutorialMap] (http://localhost:3000/api/tutorials-map)
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100): 48.933ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=2b5a0644-8b93-4dba-ac5c-7843de4b1289): 37.374ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=0af4d72f-51dd-4dc5-85e4-67c68030ce8d): 33.722ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=8bd94156-cf0a-4f88-b5f5-d2ffa57f6402): 38.009ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=e48563d4-00d1-4c0b-a5da-13e276ab119c): 44.829ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=75bb9296-0364-4ec4-8fbd-3930c9e947d6): 45.542ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=216b3270-fc89-4ad5-9053-603be0df67e8): 50.461ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=95a7e36f-44ac-4e39-9fb3-c25db26f4af9): 39.352ms
      [lib/learn-client] (GET https://ycmsw3yg4a.us-east-1.awsapprunner.com/tutorials/?limit=100&after=6d51c189-d38e-493f-b373-7e2dfe6bdbc5): 34.98ms
    [getTutorialMap] (http://localhost:3000/api/tutorials-map): 860.337ms
  [docs-view/server::prepareNavDataForClient]: 863.274ms
[[productSlug]/docs/index::getStaticProps] (consul): 2.099s
```